cmd/runtimetest/main.go

```diff
@@ -191,6 +192,81 @@ func validateSysctls(spec *rspec.Spec) error {
 	return nil
 }
 
+func mountMatch(specMount rspec.Mount, sysMount rspec.Mount) error {
+	if specMount.Destination != sysMount.Destination {
+		return fmt.Errorf("mount destination expected: %v, actual: %v", specMount.Destination, sysMount.Destination)
+	}
+
+	if specMount.Type != sysMount.Type {
+		return fmt.Errorf("mount %v type expected: %v, actual: %v", specMount.Destination, specMount.Type, sysMount.Type)
+	}
+
+	if specMount.Source != sysMount.Source {
+		return fmt.Errorf("mount %v source expected: %v, actual: %v", specMount.Destination, specMount.Source, sysMount.Source)
+	}
+
+	optMap := make(map[string]bool)
+	for _, opt := range sysMount.Options {
+		optMap[opt] = true
+	}
+	for _, option := range specMount.Options {
+		if _, ok := optMap[option]; !ok {
+			return fmt.Errorf("mount %v option %v does not exist", specMount.Destination, option)
+		}
+	}
+
+	return nil
+}
+
+func containString(opts []string, str string) bool {
+	for _, opt := range opts {
+		if opt == str {
+			return true
+		}
+	}
+
+	return false
+}
+
+func validateMounts(spec *rspec.Spec) error {
+	fmt.Println("validating mounts")
+	infos, err := mount.GetMounts()
+	if err != nil {
+		return err
+	}
+
+	mountsMap := make(map[string]rspec.Mount)
+	for _, info := range infos {
+		opts := strings.Split(info.Opts, ",")
+
+		if info.Mountpoint == "/" {
+			if spec.Root.Readonly && !containString(opts, "ro") ||
+				!spec.Root.Readonly && !containString(opts, "rw") {
+				return fmt.Errorf("root readonly expected: %v, actual: %v", spec.Root.Readonly, !spec.Root.Readonly)
+			}
+		}
+
+		mountsMap[info.Mountpoint] = rspec.Mount{
+			Destination: info.Mountpoint,
```

Comment:
This “destination → config” mapping isn't going to work if folks define several mounts over the same location.  I'm not sure why folks would want to do that (although it's close to [this workflow][1]), but it is certainly legal with the current spec wording.

[1]: http://man7.org/linux/man-pages/man8/switch_root.8.html#NOTES